server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -516,6 +538,43 @@ public final DynamicTemplate findDynamicTemplate(String fieldName, DynamicTempla
         return null;
     }
 
+    public void postProcessDynamicMappers() {
```

Comment:
It seems to me that this could live on the tail end of `parseNonDynamicArray`?

There is the 
```
assert token.isValue();
parseValue(context, lastFieldName);
```

I don't think we can get away from parsing the field more than once. But the nice thing is that this is only a cost when dynamically adding the mappings. So, parsing more than once is OK.


But having each field parsed separately is nice. It allows for a separation of concerns and for the dynamic mapping creator to create the mappings.

Reply:
I moved the call as is for now, I'm still working out how big the change is to make dims nullable and whether that should go in this PR or not. WDYT?